app/src/main/java/ch/epfl/sdp/drone3d/storage/dao/FirebaseMappingMissionDao.kt

```diff
@@ -0,0 +1,231 @@
+package ch.epfl.sdp.drone3d.storage.dao
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import ch.epfl.sdp.drone3d.storage.data.MappingMission
+import ch.epfl.sdp.drone3d.storage.data.State
+import com.google.firebase.database.*
+import com.google.firebase.database.ktx.database
+import com.google.firebase.database.ktx.getValue
+import com.google.firebase.ktx.Firebase
+import javax.inject.Inject
+
+
+class FirebaseMappingMissionDao :MappingMissionDao {
+
+    private lateinit var database: FirebaseDatabase
+
+    /**
+     * Default constructor, injected through hilt
+     */
+    @Inject
+    constructor()
+
+    /**
+     * Constructor to specify the firebase database to use.
+     *
+     * This is mostly for testing purpose
+     */
+    constructor(database: FirebaseDatabase) : this() {
+        this.database = database
+    }
+
+    private fun getDatabase(): FirebaseDatabase {
+        // Query the database if it is not initialized yet
+        if (!this::database.isInitialized)
+            database = Firebase.database("https://drone3d-6819a-default-rtdb.europe-west1.firebasedatabase.app/")//.setPersistenceEnabled(true)
+        return database
+    }
+
+    private val privateMappingMissions: MutableLiveData<List<MappingMission>> = MutableLiveData()
+    private var privateMappingMissionsIsInit: Boolean = false
+    private val sharedMappingMissions: MutableLiveData<List<MappingMission>> = MutableLiveData()
+    private var sharedMappingMissionsIsInit: Boolean = false
+
+    companion object {
+        private const val TAG = "FirebaseMappingMissionDao"
+        private const val PRIVATE_ID_PATH = "privateId"
+        private const val SHARED_ID_PATH = "sharedId"
+        private const val STATE_PATH = "state"
+        private const val MAPPING_MISSIONS_PATH = "mappingMissions"
+    }
+
+    private fun privateMappingMissionRef(UID: String): DatabaseReference {
+        return getDatabase().getReference("users/$UID/$MAPPING_MISSIONS_PATH")
+    }
+
+    private fun sharedMappingMissionRef(): DatabaseReference {
+        return getDatabase().getReference(MAPPING_MISSIONS_PATH)
+    }
+
+    private fun getMappingMission(ownerUid: String?, id: String):LiveData<MappingMission>{
```

Comment:
Could add documentation to private methods

Reply:
Done